socalabs-sid: init at 1.1.0

l1npengtul opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Socalabs is an open source plugin maker, this adds their Commodore 64 SID chip emulation plugin to nixpkgs.

https://socalabs.com/synths/commodore-64-sid/

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

l1npengtul commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>socalabs-sid</li>
  </ul>
</details>

l1npengtul commented:
cc: @magnetophon 
A non-airwindows one this time
(thanks in advance)

magnetophon commented:
This builds and runs for me, but the standalone doesn't seem to have midi input, ar at least they don't show up in pipewire.

magnetophon commented:
In ardour the vst3 has no audio out!

l1npengtul commented:
> This builds and runs for me, but the standalone doesn't seem to have midi input, or at least they don't show up in pipewire.

This is actually a bug in JUCE, and is present with many other JUCE plugins in nixpkgs! (surgeXT and odin2 were the ones I tested) 
I've applied a custom patch that should fix this, and I'm wondering if I should go ahead and patch all the other packages to include this patch.

> In ardour the vst3 has no audio out!

Fixed by changing from make to ninja. 

Also: removed redundant LTO and CMake flags, and added libXext to LDFLAGS.

l1npengtul commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>socalabs-sid</li>
  </ul>
</details>

magnetophon reviewed (approved):
It works in Ardour now, but in carla, I get an error.
```
carla-single vst3  results/socalabs-sid-x86_64-linux/lib/vst3/SID.vst3/
['/nix/store/las65hr15vcnd6l2bnywg1ribhd2ady1-carla-2.5.9/lib/carla/carla-bridge-native', 'vst3', 'results/socalabs-sid-x86_64-linux/lib/vst3/SID.vst3/', '(none)', '0']
libjack.so.0 loaded successfully!
JUCE Assertion failure in juce_VST3PluginFormat.cpp:999
JUCE Assertion failure in juce_VST3PluginFormat.cpp:3802
[carla] Plugin failed to load, error was:
Failed to get plugin description
```

The standalone works as expected.

l1npengtul commented:
> It works in Ardour now, but in carla, I get an error.
> 
> ```
> carla-single vst3  results/socalabs-sid-x86_64-linux/lib/vst3/SID.vst3/
> ['/nix/store/las65hr15vcnd6l2bnywg1ribhd2ady1-carla-2.5.9/lib/carla/carla-bridge-native', 'vst3', 'results/socalabs-sid-x86_64-linux/lib/vst3/SID.vst3/', '(none)', '0']
> libjack.so.0 loaded successfully!
> JUCE Assertion failure in juce_VST3PluginFormat.cpp:999
> JUCE Assertion failure in juce_VST3PluginFormat.cpp:3802
> [carla] Plugin failed to load, error was:
> Failed to get plugin description
> ```
> 
> The standalone works as expected.

![image](https://github.com/user-attachments/assets/45da143a-6fdd-4d4f-9524-9174f59a414a)

I can not reproduce this issue.

l1npengtul commented:
After this is merged, I plan to open PRs for the rest of the socalabs plugins.

l1npengtul commented:
Squashing commits.

l1npengtul committed:
socalabs-sid: init at 1.1.0

nbraud commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 372081`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>socalabs-sid</li>
  </ul>
</details>

l1npengtul commented:
Since there are many more socalabs plugins and I plan to package them all, I might close this PR and write one with multiple outputs. 

I'll close this PR for now, and I'll reopen if need be.

l1npengtul commented:
Never mind its a stupid idea :p.

l1npengtul commented:
Currently blocked on #376433.

nbraud commented:
> Currently blocked on #376433.

Oh, I didn't realise this was blocking you.  I was planning on leaving it 'till tomorrow, so people have time to react prior to merging.

l1npengtul commented:
> > Currently blocked on #376433.
> 
> Oh, I didn't realise this was blocking you.  I was planning on leaving it 'till tomorrow, so people have time to react prior to merging.

Wait I forgot it doesn't, this repo packages VST2SDK itself. oops.

feel free to take your time and apologies 😅

nbraud commented:
> feel free to take your time and apologies 😅

No problem, I was just confused since it looked like the SDK was vendored-in.

nbraud commented:
> Squashing commits.

PS: It's not necessary, when there's a single commit in the end, since squash-merge is an option too.
